### ISSUE TITLE
"Unexpected token" build error with legacy / stage 1 decorators (no `@e-d/babel-transforms`)

### ISSUE THREAD

**Issue by niranjan94** [machty/ember-concurrency-decorators#48]
Code snippet in question:

```js
export default class AppSearchComponent extends Component {

  @restartableTask
  *searchContent(keyword) {
    if (isEmpty(keyword)) {
      return [];
    }
    yield timeout(250);
    return yield this.store.query('content', { keyword });
  }

}
```

Error on build:

```
Build Error (broccoli-persistent-filter:Babel > [Babel: sandbox]) in sandbox/components/navigation/app-search.js

/Users/xyz/sandbox/app/components/navigation/app-search.js: Unexpected token (17:26)

  15 | 
  16 |   @restartableTask
> 17 |   *searchContent(keyword) {
     |                           ^
  18 |     if (isEmpty(keyword)) {
  19 |       return [];
  20 |     }
```

---

Dependencies include: 

```json
{
    "ember-concurrency": "^0.10.0",
    "ember-concurrency-decorators": "^1.0.0-beta.2",

    "ember-decorators": "^6.0.0",
    "ember-decorators-polyfill": "^1.0.4",

    "ember-cli-babel": "^7.7.3",
    "ember-cli": "~3.9.0",
    "ember-source": "~3.9.0"
}
```

And not using `@ember-decorators/babel-transforms`

**Comment by buschtoens** [comment]
This is caused by https://github.com/babel/babel/issues/9852.

You can either:

- wait until https://github.com/babel/babel/pull/9912 is merged and released
- pre-publish https://github.com/babel/babel/pull/9912 and force a [yarn resolution](https://yarnpkg.com/en/docs/selective-version-resolutions/)
- use the assignment syntax that is shown in the [`README.md`](https://github.com/machty/ember-concurrency-decorators#task), e.g.:
  ```ts
  export default class ExampleComponent extends Component {
    @task
    doStuff = function*() {
      // ...
    };
  }
  ```
- keep using v5 of `ember-decorators` plus `@ember-decorators/babel-transforms` to get stage 2 decorators (_not recommended_)

**Comment by niranjan94** [comment]
@buschtoens thanks for the update 😄 Will wait for https://github.com/babel/babel/pull/9912 to be merged. Using the assignment syntax in the meantime.

**Comment by buschtoens** [comment]
I just merged #54 and released it as [`v1.0.0-beta.3`](https://github.com/machty/ember-concurrency-decorators/releases/tag/v1.0.0-beta.3). If you manually apply the patch to `@babel/parser` as described in [_What happened to the fancy generator method syntax?_](https://github.com/machty/ember-concurrency-decorators#what-happened-to-the-fancy-generator-method-syntax) or wait for it to be released officially, you can use the generator method syntax again.

**Comment by gabrielgrant** [comment]
@buschtoens looks like [the latest Babel release (7 days ago) is v7.5.5](https://www.npmjs.com/package/@babel/core), which presumably includes https://github.com/babel/babel/pull/9912 (merged 24 days ago). Does that mean this patch work-around is no longer needed in order to use the "fancy generator method syntax" and the note can be removed/examples updated?

**Comment by buschtoens** [comment]
@gabrielgrant Yes. The fix was already released in [`v7.5.0`](https://github.com/babel/babel/releases/tag/v7.5.0). So, if you're on at least that version, everything should be groovy.

Also, the version of `@babel/core` in `ember-cli` itself doesn't actually matter. It's the version in [`ember-cli-babel`](https://github.com/babel/ember-cli-babel/blob/0b5f29934104f277e09b89002a254ce07b42f532/package.json#L43-L45), which is a dependency in your _project's_ `package.json`.

If you are using yarn, you can force an upgrade of this `@babel/core` dependency by setting a [resolution](https://yarnpkg.com/lang/en/docs/selective-version-resolutions/). If you're using npm or don't want to use a resolution, a clean re-installation of the dependencies (delete `node_modules` and any package manager lock files) should also do the trick.
